.github/workflows/client.yml

```
@@ -0,0 +1,42 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows 
+on:  
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true # ignore the failure of a step and avoid terminating the job
+      - name: Run Docker build
+        run: |
+          docker build \
+            --target gobuild \
+            --tag go-build-env .
+          docker build \
+            --tag keep-client . 
+      - name: Create and enter test results directory
```

We're not changing a directory, right?
```suggestion
      - name: Create test results directory
```